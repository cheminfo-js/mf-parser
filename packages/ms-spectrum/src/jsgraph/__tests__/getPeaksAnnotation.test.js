'use strict';

const getPeaksAnnotation = require('../getPeaksAnnotation');

describe('test getPeaksAnnotation', () => {
  let peaks = [
    { x: 12, y: 1 },
    { x: 24, y: 2 },
    { x: 36, y: 3 },
    { x: 42, y: 4 }
  ];
  it('default options', () => {
    let result = getPeaksAnnotation(peaks);
    expect(result).toMatchSnapshot();
  });
});
