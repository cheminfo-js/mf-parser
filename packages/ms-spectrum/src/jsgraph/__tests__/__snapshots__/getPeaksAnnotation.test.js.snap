// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test getPeaksAnnotation default options 1`] = `
Array [
  Object {
    "_highlight": undefined,
    "labels": Array [
      Object {
        "position": Object {
          "dx": "2px",
          "dy": "-17px",
          "x": 12,
          "y": 1,
        },
        "text": "12.00000",
      },
    ],
    "position": Array [
      Object {
        "dy": "-5px",
        "x": 12,
        "y": 1,
      },
      Object {
        "dy": "-25px",
        "x": 12,
        "y": 1,
      },
    ],
    "type": "line",
  },
  Object {
    "_highlight": undefined,
    "labels": Array [
      Object {
        "position": Object {
          "dx": "2px",
          "dy": "-17px",
          "x": 24,
          "y": 2,
        },
        "text": "24.00000",
      },
    ],
    "position": Array [
      Object {
        "dy": "-5px",
        "x": 24,
        "y": 2,
      },
      Object {
        "dy": "-25px",
        "x": 24,
        "y": 2,
      },
    ],
    "type": "line",
  },
  Object {
    "_highlight": undefined,
    "labels": Array [
      Object {
        "position": Object {
          "dx": "2px",
          "dy": "-17px",
          "x": 36,
          "y": 3,
        },
        "text": "36.00000",
      },
    ],
    "position": Array [
      Object {
        "dy": "-5px",
        "x": 36,
        "y": 3,
      },
      Object {
        "dy": "-25px",
        "x": 36,
        "y": 3,
      },
    ],
    "type": "line",
  },
  Object {
    "_highlight": undefined,
    "labels": Array [
      Object {
        "position": Object {
          "dx": "2px",
          "dy": "-17px",
          "x": 42,
          "y": 4,
        },
        "text": "42.00000",
      },
    ],
    "position": Array [
      Object {
        "dy": "-5px",
        "x": 42,
        "y": 4,
      },
      Object {
        "dy": "-25px",
        "x": 42,
        "y": 4,
      },
    ],
    "type": "line",
  },
]
`;
